arlo_server/all_routes/election/__element__.py

```diff
@@ -0,0 +1,91 @@
+from flask import jsonify, Response, request
+from jsonschema import validate
+
+from arlo_server import app, db
+from arlo_server.auth import require_audit_admin_for_organization
+from arlo_server.models import Election
+
+ELECTION_NAME_KEY = "electionName"
+ELECTION_ONLINE_KEY = "online"
+ELECTION_RANDOM_SEED_KEY = "randomSeed"
+ELECTION_RISK_LIMIT_KEY = "riskLimit"
+ELECTION_STATE_KEY = "state"
+
+PUT_ELECTION_REQUEST_SCHEMA = {
+    "type": "object",
+    "properties": {
+        ELECTION_NAME_KEY: {"type": "string"},
+        ELECTION_ONLINE_KEY: {"type": "boolean"},
+        ELECTION_RANDOM_SEED_KEY: {"type": "string"},
+        ELECTION_RISK_LIMIT_KEY: {"type": "integer"},
+        ELECTION_STATE_KEY: {"type": "string"},
+    },
+    "additionalProperties": False,
+}
+
+GET_ELECTION_RESPONSE_SCHEMA = {
+    "type": "object",
+    "properties": {
+        "election": {
+            "type": "object",
+            "properties": {
+                ELECTION_NAME_KEY: {"type": "string"},
+                ELECTION_ONLINE_KEY: {"type": "boolean"},
+                ELECTION_RANDOM_SEED_KEY: {"type": "string"},
+                ELECTION_RISK_LIMIT_KEY: {"type": "integer"},
+                ELECTION_STATE_KEY: {"type": "string"},
+            },
+            "additionalProperties": False,
+        },
+    },
+    "additionalProperties": False,
+}
+
+
+@app.route("/election/<election_id>", methods=["GET"])
+def get_election(election_id: str) -> Response:
+    election = Election.query.filter_by(id=election_id).one()
+    require_audit_admin_for_organization(election.organization_id)
+
+    response_data = {
+        "election": {
+            ELECTION_NAME_KEY: election.name,
+            ELECTION_ONLINE_KEY: election.online,
+            ELECTION_RANDOM_SEED_KEY: election.random_seed,
+            ELECTION_RISK_LIMIT_KEY: election.risk_limit,
+            ELECTION_STATE_KEY: election.state,
+        }
+    }
+
+    validate(schema=GET_ELECTION_RESPONSE_SCHEMA, instance=response_data)
+
+    return jsonify(response_data)
+
+
+@app.route("/election/<election_id>", methods=["PUT"])
+def put_election(election_id: str) -> Response:
+    election = Election.query.filter_by(id=election_id).one()
```

Comment:
I just found `Election.query.get_or_404(election_id)`. What do you think of starting to use that? https://flask-sqlalchemy.palletsprojects.com/en/2.x/queries/#queries-in-views

Reply:
Another approach would be at add this to `handle_500`:

```py
    if isinstance(original, NoResultFound):
        return (
            jsonify(
                errors=[{"message": str(original), "errorType": NotFound.__name__}]
            ),
            NotFound.code,
        )
```

The advantage is we can just keep using `.one()` and the like and it'll just work. The disadvantage is we can't easily change the error message which would look like this:

```
{"errors":[{"errorType":"NoResultFound","message":"No row was found for one()"}]}
```

Reply:
Why couldn't we change the error message? Couldn't we just change the "message" property in the snippet you wrote?

I am torn. That is very clever and I like it a lot. I also value being explicit and minimizing magic.

Reply:
> Why couldn't we change the error message? Couldn't we just change the "message" property in the snippet you wrote?

We could, but that message is generic for every `NoResultFound` so we'd have to use the same message for everything unless the exception gives us enough information to figure out what happened.

Anyway, I updated the PR to use `get_or_404`.